api/v1alpha4/indexer.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// MachineNodeNameIndex is used by the Machine Controller to index Machines by Node name, and add a watch on Nodes.
+	MachineNodeNameIndex = "status.nodeRef.name"
+)
+
+// IndexMachineByNodeName generates the extractVaue for the node name
+// machine index for the cache.
+func IndexMachineByNodeName(o client.Object) []string {
```

Comment:
can we move this logic to noderefutil as well and drop this import to "sigs.k8s.io/controller-runtime/pkg/client" from `api/`?

Reply:
@enxebre Good point. I have now refactored the PR so that `api` does not import the `sigs.k8s.io/controller-runtime/pkg/client` package but the information about the index that is derived from the Machine type is still in the `api` package.